extensions/amp-story/1.0/amp-story-tooltip.js

```diff
@@ -202,46 +239,71 @@ export class AmpStoryTooltip {
   }
 
   /**
-   * Builds tooltip and attaches it depending on the clicked element content and
+   * Builds tooltip and attaches it depending on the target's content and
    * position.
-   * @param {!Element} clickedEl
+   * @param {!Element} target
    * @private
    */
-  attachTooltipToEl_(clickedEl) {
-    const elUrl = clickedEl.getAttribute('href');
-    if (!isProtocolValid(elUrl)) {
-      user().error(TAG, 'The tooltip url is invalid');
-      return;
-    }
+  updateTooltipEl_(target) {
+    this.updateTooltipText_(target);
+    this.updateTooltipIcon_(target);
+    this.positionTooltip_(target);
+  }
 
-    const iconUrl = clickedEl.getAttribute('data-tooltip-icon');
-    if (!isProtocolValid(iconUrl)) {
-      user().error(TAG, 'The tooltip icon url is invalid');
-      return;
+  /**
+   * Updates tooltip behavior depending on the target.
+   * @param {!Element} target
+   */
+  updateTooltipBehavior_(target) {
+    if (matches(target, tooltipDelegatableSelectors().LINK)) {
+      addAttributesToElement(dev().assertElement(this.tooltip_),
+          dict({'href': this.getElementHref_(target)}));
     }
+    else if (matches(target, tooltipDelegatableSelectors().TWITTER)) {
+      this.tooltip_.addEventListener('click',
+          this.expandComponentHandler_, true);
+    }
+  }
 
-    const {href} = parseUrlDeprecated(elUrl);
-
-    const tooltipText = clickedEl.getAttribute('data-tooltip-text') ||
-      getSourceOriginForElement(clickedEl, href);
-    this.updateTooltipText_(tooltipText);
+  /**
+   * Handles the event of an interactive element coming into expanded view.
+   * @param {!Event} event
+   * @private
+   */
+  onExpandComponent_(event) {
+    event.preventDefault();
+    event.stopPropagation();
 
-    const iconSrc = iconUrl ? parseUrlDeprecated(iconUrl).href :
-      DEFAULT_ICON_SRC;
-    this.updateTooltipIcon_(iconSrc);
+    this.closeTooltip_();
+    this.previousTarget_.classList
+        .toggle('i-amphtml-expandable-component-shield', false);
+    this.storeService_.dispatch(
+        Action.TOGGLE_EXPANDED_COMPONENT, this.previousTarget_);
+  }
 
-    addAttributesToElement(dev().assertElement(this.tooltip_),
-        dict({'href': href}));
+  /**
+   * Gets href from an element containing a url.
+   * @param {!Element} target
+   * @private
+   */
+  getElementHref_(target) {
+    const elUrl = target.getAttribute('href');
+    if (!isProtocolValid(elUrl)) {
+      user().error(TAG, 'The tooltip url is invalid');
+      return;
+    }
 
-    this.positionTooltip_(clickedEl);
+    return parseUrlDeprecated(elUrl).href;
   }
 
   /**
    * Updates tooltip text content.
-   * @param {string} tooltipText
+   * @param {!Element} target
    * @private
    */
-  updateTooltipText_(tooltipText) {
+  updateTooltipText_(target) {
+    const tooltipText = target.getAttribute('data-tooltip-text') ||
+      getSourceOriginForElement(target, this.getElementHref_(target));
```

Comment:
What happens to the UI if the url is invalid?

Reply:
It will display the origin's domain. e.g. if `href=javascript:alert('hello world')`, then `localhost` will be displayed (with an error in the console).